### ISSUE TITLE
Convert mascot crosslink search to ssl

### ISSUE THREAD

**Issue by magdiws** [Rappsilber-Laboratory/xiView_container#73]
Hi,

I am trying to import a cross-link search from Mascot into xiVIEW and convert the data to an ssl file to create a library for searching files in Skyline. If I only import the xiVIW-csv into xiVIW I don’t get the option to export to an ssl file in the export drop-down menu. If I try to import both xiVIEW-csv, mgf and fasta files the mgf gives me an error. I use the same mgf file as I used for the Mascot search and I have also tried to use an mgf file generated from the search in Mascot but both gives me the following error message:

log id: b0s00djk16o01dac9evcorrc14_1596713464.log (include this in the github issue)
error type: Error
message: Error reading peak list file F075722.mgf: IOError - Arguments:
(2, 'No such file or directory')
id: undefined

**Comment by colin-combe** [comment]
Hi,
sorry for the delay.

It looks like the CSV file you uploaded refers to an .mgf file that has a different name from the one you uploaded. It could also be a problem on our side. If you're OK with sending me the files I can check. 

Also, from our email, I see your using the Mascot instructions which say to upload a CSV file. xiVIEW works better if provided with an mzIdentML file, which is a file format Mascot also exports I think. However, I don't know if Mascot's mzIdentML export works correctly with crosslinks - I'd be interested know. Its something you could maybe try and let the know the results or let me see the file.

Lastly, the SSL export... if I remember rightly this function was never in the 'public' version of xiVIEW but only in my lab's version, which is slightly different because its hooked up to a different database. I have a bit of work to do to make .ssl export work in the public version. I ~think~ you don't need to upload a peaklist to have the information needed to create the .ssl file. Its just something I need to get around to fixing for the public version.

So basically, you've got a little wait before the .ssl export appears in the public version - i'll try to get round to it soon(ish).

We can look into the mgf file issue if you want, and also I'd like to know what happens re. the Mascot mzIdentML file. 

If the Mascot mzIdentML file works it should contain all the info for generating an .ssl file. The CSV file won't and you'll need to enter some info manually. So getting the mzIdentML file to work could be seen as a step towards .ssl generation ;)

**Comment by magdiws** [comment]
Hi Colin,

Thank you very much for your response. Changing the .mgf file name to the one Mascot assigned it solved the importing issue. 

Regarding Mascot mzIdentML file, unfortunately Mascot only allows exporting crosslinking results to 4 file formats: FASTA, xiView-CSV, Mascot DAT file and MGF peak list at the moment. So if I understand you correctly I will not be able to export to ssl before Mascot enables exporting crosslinking data to mzIdentML, is that correct?

**Comment by colin-combe** [comment]
that's not completely correct - I would prefer to base things on an uploaded mzIdentML file, but it could be done by uploading a CSV file and then entering info such as modification masses manually. tbh i can't remember what info's needed to generate an .ssl file at the moment, its a while since I looked at it.

It's not a huge task to fix it, but there are other things on my todo list. As you will have gathered from the emails, one of the authors of xiVIEW has left the group, so I've inherited responsibility for maintaining some code I didn't write.

In short, watch this space but i don't have a time estimate for you. (I'll also email you when its fixed.)

Glad the peak list upload is working, seeing the annotated spectra is one of the main points of xiVIEW.

**Comment by magdiws** [comment]
Ok, I understand. Thank you for your support and looking forward to using the application.

**Comment by colin-combe** [comment]
as an aside:
> Regarding Mascot mzIdentML file, unfortunately Mascot only allows exporting crosslinking results to 4 file formats: FASTA, xiView-CSV, Mascot DAT file and MGF peak list at the moment.

Another reason to take up mzIdentML export of crosslinks with the Mascot developers is that this allows you to submit the dtaa to the Jpost Proteome exchange repository and get 'complete submission' status for it.
(Its currently the only way to get this gold star for crosslink data.)

**Comment by chambm** [comment]
Hi @colin-combe , how does the SSL export work? We didn't explicitly add support to that format for crosslinking, although we did add one-line-style crosslinking support to the BiblioSpec format itself back in May.

**Comment by chambm** [comment]
Ping!

**Comment by colin-combe** [comment]
>  how does the SSL export work?

it doesn't

it nearly works by concatenating the two crosslinked peptides together with an AA and a modification that represent the crosslink, to form single peptide that acts as a proxy for the two crosslinked ones, and putting this in the .ssl file

**Comment by colin-combe** [comment]
@chambm - i'll let you know when its fixed if you want to check it out

**Comment by chambm** [comment]
OK, that's probably for the best. We added support in BiblioSpec and Skyline for a one-line, 2-peptides-crosslinked format. It supports either crosslinks or looplinks. A crosslink looks like:
`PEPTIDEA-PEPTIDEB-[+linkermMass@positionA-positionB]`
A looplink looks like:
`PEPTIDE-[+linkerMass@positionA1,positionA2]` (note the comma instead of dash)
Do you think we could unify on this representation? We'd add support in BiblioSpec's own Mascot reader to get the same kind of format out.
